service/insolvency_service.go

```diff
@@ -0,0 +1,36 @@
+package service
+
+import (
+	"fmt"
+
+	"github.com/companieshouse/insolvency-api/models"
+
+	"github.com/companieshouse/chs.go/log"
+	"github.com/companieshouse/insolvency-api/dao"
+)
+
+// ValidateInsolvencyDetails checks that an insolvency case is valid and ready for submission which is returned as a boolean
+// Any validation errors found are added to an array to be returned
+func ValidateInsolvencyDetails(svc dao.Service, transactionID string) (bool, *[]models.ValidationErrorResponseResource) {
+
+	validationErrors := make([]models.ValidationErrorResponseResource, 0)
+
+	// Retrieve details for the insolvency resource from MongoDB
+	_, err := svc.GetInsolvencyResource(transactionID)
```

Comment:
Should there also be a check to see if the Insolvency resource doesn't exist in Mongo?

Reply:
If it doesn't exist in Mongo it'll return an error that gets picked up below